lib/puppet/pops/evaluator/compare_operator.rb

```diff
@@ -222,7 +222,7 @@ def match_Hash(hash, left, scope)
 
   def match_Symbol(symbol, left, scope)
     return true if symbol == :default
-    equals(left, default, scope)
+    match(left, default, scope)
```

Comment:
I think the code is broken in other ways - there does not seem to be a value associated with `default` local variable - there is no such method either. I think this is a hidden bug because we never run into the situation that the evaluator ever matches/compares against any other symbol than `:default`. If it were, it should probably do an `equal`, but then doing an equals of `symbol` and `left`.

The only way to trigger the faulty original logic would be if there is faulty AST trying to compare some other symbol than `:default`.

I think the correct change here is to change line 225 to `equals(symbol, left, scope)` as that would make it correctly handle the `:undef` symbol that could be to only possible other relative thing to compare against. 

Ping @thallgren for second opinion.